### PR TITLE
backports: for v0.5.3

### PR DESCRIPTION
**Pull request by smira**
PRs backported:

- #908
- #909 
- #910

### CODE REVIEW

File: sfyra/pkg/tests/tests.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func Run(ctx context.Context, cluster talos.Cluster, vmSet *vm.Set, capiManager
 		},
 		{
 			"TestCompatibilityCluster",
-			TestCompatibilityCluster(ctx, metalClient, cluster, vmSet, capiManager, options.PrevTalosRelease, options.KubernetesVersion),
+			TestCompatibilityCluster(ctx, metalClient, cluster, vmSet, capiManager, options.PrevTalosRelease, "v1.22.12"),
```
Comment (frezbo) [inline sfyra/pkg/tests/tests.go:99]: shouldn't the k8s version come from some config?

**Review by frezbo — APPROVED** [review]

File: sfyra/pkg/tests/tests.go (hunk 1 above)
Comment (smira) [reply sfyra/pkg/tests/tests.go:99]: Yes, it does, but this test specifically boots old Talos, so we need old Kubernetes

Comment (smira) [reply sfyra/pkg/tests/tests.go:99]: so the long story was that previously we used same version for "old" Talos (0.13) and "new" Talos (0.14+). As we're right now on Talos 1.1, which is Kubernetes 1.24, it's no longer compatible with 0.13, so I have to pin old version. Hopefully we should be able to remove this test sooner or later, as it's only needed for Talos 0.13 which doesn't have SideroLink (so it's based on init nodes).

**Pull request merged**